pkg/certificate/providers/mrc.go

```diff
@@ -0,0 +1,118 @@
+package providers
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/openservicemesh/osm/pkg/apis/config/v1alpha2"
+	"github.com/openservicemesh/osm/pkg/certificate"
+	"github.com/openservicemesh/osm/pkg/k8s/informers"
+)
+
+var (
+	errNilInformerCollection = errors.New("informer collection is nil")
+)
+
+var mrcKeyFunc = func(mrc v1alpha2.MeshRootCertificate) string {
+	return fmt.Sprintf("%s/%s", mrc.GetNamespace(), mrc.GetName())
+}
+
+// MRCComposer is a composer object that allows consumers
+// to observe MRCs (via List() and Watch()) as well as generate
+// `certificate.Provider`s from those MRCs
+type MRCComposer struct {
```

Comment:
Can we get rid of the MRCComposer, and put these methods on the MRC client?

Reply:
`MRCClient` is an interface; are you suggesting turning it into a struct?

Reply:
err.. sorry i was getting mixed up with the compatclient vs this vs interface.